src/rgw/rgw_auth.cc

```diff
@@ -323,10 +323,22 @@ rgw::auth::Strategy::apply(const DoutPrefixProvider *dpp, const rgw::auth::Strat
     } catch (const int err) {
       ldpp_dout(dpp, 5) << "applier throwed err=" << err << dendl;
       return err;
+    } catch (const std::exception& e) {
+      ldpp_dout(dpp, 5) << "applier throwed unexpected err: " << e.what()
+                        << dendl;
+      return -EPERM;
+    } catch (...) {
```

Comment:
The ... catch shouldn't be necessary, since std::exception should catch everything, right?

Reply:
Shouldn't be in general unless there's some exception class that doesn't inherit from std::exception (I don't know if there's any such class involved here).

Reply:
I checked the code here again. Actually, there would be two kinds of errors: int and std::exception, so i remove the part of `catch (...)`.